### DISCUSSION TITLE
Omitting `Exception` suffix when throwing exceptions, like when declaring attributes

### DISCUSSION THREAD

**Discussion by Rekkonnect** [dotnet/csharplang#5204] [Language Ideas]
That's a discussion that I'd like to see opinions on, what if when throwing exceptions, you could just write `throw new NullReference(...)`? Or `throw new DivideByZero(...)`? Some exceptions won't look nice without the suffix (like `Argument(Exception)`).

The way exception types are identified should work the same as attributes when being declared. Outside of the context of throwing an exception, simplified exception names are not permitted or resolved, like attribute types.

**Comment by svick** [comment]
Personally, I don't think this reads well. For example, "throw new null reference" doesn't make sense, you're not throwing a reference.

Contrast this with attributes, where something like `[Serializable] class Foo` reads fine as "serializable class Foo".

Of course, this simplistic way of reading C# code as English phrases is not going to work well for anything more complex, and C# definitely isn't trying to follow English grammar (e.g. contrast LINQ with SQL). But I do think it can be a useful guide in this case.

**Reply by DavidArno** [reply]
I disagree to be honest. `throw new NullReference(...)` actually makes more sense to me than `throw new NullReferenceException(...)`. It's self-evident that we are handling an exception and that exception is due to a null reference, so the `Exception` suffix is redundant. `NullReference` tells me what the problem is.

But judging by the votes for your answer, I'm in the minority here. And in reality, this boat sailed years ago. Whilst I'd have preferred that `Exception` suffix to have been treated like the `Attribute` suffix, it wasn't and I'm not sure it makes sense to change it now.

**Comment by deeprobin** [comment]
But what if you want to throw `Exception`?
```cs
throw new Exception("");
```

**Reply by jnm2** [reply]
@deeprobin Under this proposal, then you would write `throw new Exception("")`.

**Reply by DaZombieKiller** [reply]
With target-typed `new`, you can actually do
```cs
throw new("");
```
today, which will throw an `Exception`.

**Comment by VBAndCs** [comment]
You can do it yourself  be writing a Roslyn source generator, to generate an ArgumentNull class when it appears in a throw or catch statements, and just make this class inherit ArgumentNullException.
`public ArgumentNull : ArgumentNullException {}`

**Reply by CyrusNajmabadi** [reply]
this won't work as `catch (ArgumentNull)` will not catch `ArgumentNullException`.

**Reply by jnm2** [reply]
You could write a Roslyn source generator that generates:

```cs
global using ArgumentNull = System.ArgumentNullException;
```

Unfortunately, all exception types will then be in scope everywhere regardless of whether their original namespace is imported, and they won't *look* like exception types in intellisense.
